web/packages/teleterm/src/ui/DocumentTerminal/DocumentTerminal.tsx

```diff
@@ -118,9 +117,16 @@ export function DocumentTerminal(props: {
       autoFocusDisabled={true}
     >
       {$fileTransfer}
-      {ptyProcess && (
+      {attempt.status === 'success' && (
```

Comment:
An example of why not checking the `status` before accessing `attempt.data` is dangerous.

When the `run` callback from `useAsync` is called again, `status` goes from `success` to `processing`. However, `attempt.data` does not get cleanup up!

This is useful in some cases, for example if you want to show stale state while loading fresh state. But the downside to it is that if you don't account for `attempt.status`, you might end up using stale values.

---

Here when I was initially working on this PR, I couldn't understand why we call stuff on the stale `ptyProcess`. Turns out this line was causing this.